consumer/consumererror/throttle.go

```diff
@@ -0,0 +1,46 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package consumererror // import "go.opentelemetry.io/collector/consumer/consumererror"
+
+import (
+	"time"
+)
+
+// TODO: Clean this by forcing all exporters to return an internal error type that always include the information about retries.
+
+type Throttle struct {
+	err   error
+	delay time.Duration
+}
+
+func (t Throttle) Error() string {
+	return "Throttle (" + t.delay.String() + "), error: " + t.err.Error()
+}
+
+func (t Throttle) Unwrap() error {
+	return t.err
+}
+
+func (t Throttle) RetryDelay() time.Duration {
+	return t.delay
+}
+
+// NewThrottleRetry creates a new throttle retry error.
+func NewThrottleRetry(err error, delay time.Duration) error {
```

Comment:
Do we do wrapping here or ask to use something like `multierr.Append`?

Reply:
We are wrapping.  Throttle has an Unwrap() function.  Users don't need to use multierr.

Reply:
I understand what we are doing, sorry for the wrong question. Was trying to ask if we should do automatic wrapping or ask users to use something like `multierr.Append`?

Reply:
I think we should automatically wrap here.  The expectation is that the error passed is based on a single request, rather than a multierr error, so I think wrapping is correct here.  If there are multiple things that need to be retried, those could be combined via multierr.Append and then potentially un-combined when we need to find the longest retry time, but we should still be wrapping in this case.

Reply:
> The expectation is that the error passed is based on a single request, rather than a multierr error, so I think wrapping is correct here.

It does not matter if it is one or more requests, the fact that matters is there needs to be an existing error.

Still not understand why enforce wrapping, sorry. You are saying that we cannot have a Throttle error without having another error, which I am not sure I understand.

My suggestion is that `Throttle` is an independent error, and if users want to say that `Throttle` is caused by another error they will do `multierr.Append(NewThrottle(...), myErr)` or `multierr.Combine(NewThrottle(...), myErr)`. 

I am trying to understand if we need to force users to have a previous error or not.

Reply:
I would expect the error to give some reason as to "why" we need to retry or are throttled.

If we use multierr.Append/Combine, then it isn't clear if those errors were for two different events (e.g. in the GCP exporter, maybe I split a batch of data into two, made two requests, and got two errors), or for the same event (e.g. one request with an error which I am retrying).

Reply:
OK, what is the order of arguments in other cases like this (if you seen any error like this one)? For me seems more natural to pass the delay first. Also in all `fmt.Errorf` statements I see the "cause" as last thing.

Reply:
mmm.  I looked at a few other error classes (e.g. gRPC status), and it seems better to not include other errors.  Putting the error into the throttle error doesn't actually make the error more readable, since they get printed out as a list anyways.

I simplified this by removing the error as an argument